cluster/v1alpha1/types_rolloutstrategy.go

```diff
@@ -0,0 +1,96 @@
+
+	// All RolloutStrategyType
+	// +optional
+	All *RolloutAll `json:"all,omitempty"`
+
+	// Progressive RolloutStrategyType
+	// +optional
+	Progressive *RolloutProgressive `json:"progressive,omitempty"`
+
+	// ProgressivePerGroup RolloutStrategyType
+	// +optional
+	ProgressivePerGroup *RolloutProgressivePerGroup `json:"progressivePerGroup,omitempty"`
+}
+
+// RolloutStrategy Types
+type RolloutStrategyType string
+
+const (
+	//All means apply the workload to all clusters in the decision groups at once.
+	All RolloutStrategyType = "All"
+	//Progressive means apply the workload to the selected clusters progressively per cluster.
+	Progressive RolloutStrategyType = "Progressive"
+	//ProgressivePerGroup means apply the workload to the selected clusters progressively per group.
+	ProgressivePerGroup RolloutStrategyType = "ProgressivePerGroup"
+)
+
+// RolloutAll is RolloutStrategyType
+type RolloutAll struct {
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// RolloutProgressivePerGroup is RolloutStrategyType
+type RolloutProgressivePerGroup struct {
+	// List of the decision groups names or indexes to apply the workload first and fail if workload did not reach successful ((not proceed to apply workload to other decision groups/clusters).
+	// +optional
+	MandatoryDecisionGroups []MandatoryDecisionGroup `json:"mandatoryDecisionGroups,omitempty"`
+
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// RolloutProgressive is RolloutStrategyType
+type RolloutProgressive struct {
+	// List of the decision groups names or indexes to apply the workload first and fail if workload did not reach successful ((not proceed to apply workload to other decision groups/clusters).
+	// +optional
+	MandatoryDecisionGroups []MandatoryDecisionGroup `json:"mandatoryDecisionGroups,omitempty"`
+
+	// MaxConcurrency is the max number of clusters to deploy workload concurrently. The default value for MaxConcurrency is determined from the clustersPerDecisionGroup defined in the placement->DecisionStrategy.
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +optional
+	MaxConcurrency string `json:"maxConcurrency,omitempty"`
+
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// Timeout to consider while applying the workload.
+type Timeout struct {
+	// Timeout define how long workload applier controller will wait till workload reach successful state in the cluster. Only considered for Rollout Type Progressive and ProgressivePerGroup.
+	// Timeout default value is None meaning the workload applier will not proceed apply workload to other clusters if did not reach the successful state.
+	// Timeout must be defined in [0-9h]|[0-9m]|[0-9s] format examples; 2h , 90m , 360s
+	// +kubebuilder:validation:Pattern="^(([0-9])+[h|m|s])|None$"
+	// +kubebuilder:default:=None
+	// +optional
+	Timeout string `json:"timeout,omitempty"`
+}
+
+// MandatoryDecisionGroup set the decision group name or group index.
+type MandatoryDecisionGroup struct {
```

Comment:
How to handle the case when both GroupName and GroupIndex are defined in each MandatoryDecisionGroup?